docs/fsharp/style-guide/formatting.md

```diff
@@ -1140,6 +1140,67 @@ Option.traverse(
 )
 ```
 
+## Formatting generic type arguments and constraints
+
+Note that due to current F# indentation rules, the formatting in this section violates the “avoid name sensitive alignment” clause mentioned previously in this document.
+
+The guidelines below apply to both functions, members, and type definitions.
+
+Keep generic type arguments and constraints on a single line if it’s not too long:
+
+```fsharp
+let f<'a, 'b when 'a : equality and 'b : comparison> param =
+    // function body
+```
+
+If both generic type arguments/constraints and function parameters don’t fit, but the type parameters/constraints alone do, place the parameters on new lines:
+
+```fsharp
+let f<'a, 'b when 'a : equality and 'b : comparison>
+    param
+    =
+    // function body
+```
+
+If the constraints are too long, break and align them as shown below. Keep each constraint on a single line regardless of its length. Place `>` at the end of the last line. Note that in the case of statically resolved type parameters, the required space between `<` and `^` influences the indentation.
+
+```fsharp
```

Comment:
This seems like a vanity indentation.
How about:
```fsharp
let inline f< ^a, ^b
    when ^a : (static member Foo1: unit -> ^b)
    and ^b : (member Foo2: unit -> int)
    and ^b : (member Foo3: string -> ^a option)>
    arg1
    arg2
    =
    // function body
    ()
```

Reply:
Sadly not allowed by the compiler, as mentioned at the start of the section.

Reply:
Oh wait, it is allowed! Yay! I'll update.